src/maptm/src/maptm_ovsdb.c

```diff
@@ -0,0 +1,307 @@
+        return false;
+    }
+
+    if (!(osp_ps_get(ps, "MAPT_SUPPORT", mapt_support, sizeof(mapt_support))))
+    {
+        LOGE("%s: Cannot get MAPT_SUPPORT value", __func__);
+        return false;
+    }
+
+    snprintf(mapt_support, sizeof(mapt_support), "%s", "true");
+    if (!(osp_ps_set(ps, "MAPT_SUPPORT", mapt_support, sizeof("MAPT_SUPPORT"))))
+    {
+        LOGE("%s: Cannot save MAPT_SUPPORT through osp API", __func__);
+        return false;
+    }
+    LOGT("%s: MAPT_SUPPORT = %s", __func__, mapt_support);
+
+    if(strncmp(mapt_support,"true",sizeof("true")-1))
+    {
+        strucWanConfig.mapt_support = false;
+    }
+
+    return ret;
+}
+
+// Get MAPT_Support value
+bool maptm_get_supportValue(char *value)
+{
+    char str[64];
+    const char delim[] = ":";
+
+    STRSCPY(str, value);
+
+    char *ptr = strtok(str, delim);
+
+    while (ptr != NULL)
+    {
+        if (!strcmp(ptr, "\"true\""))
+        {
+            return true;
+        }
+        else if (!strcmp(ptr, "\"false\""))
+        {
+            return false;
+        }
+        ptr = strtok(NULL, ",");
+    }
+
+    return false;
+}
+
+// Node_Config callback
+void callback_Node_Config(
+        ovsdb_update_monitor_t *mon,
+        struct schema_Node_Config *old_rec,
+        struct schema_Node_Config *conf
+)
+{
+    if ((mon->mon_type) == OVSDB_UPDATE_NEW)
+    {
+        LOGD("%s: new node config entry: module %s, key: %s, value: %s",
+                __func__, conf->module, conf->key, conf->value);
+
+        if (!strcmp(conf->module, "MAPTM"))
+        {
+            if (maptm_get_supportValue(conf->value))
+            {
+                WanConfig |= MAPTM_ELIGIBILITY_ENABLE;
+            }
+            strucWanConfig.mapt_support = maptm_get_supportValue(conf->value);
+            maptm_dhcp_option_update_15_option(strucWanConfig.mapt_support);
+            maptm_dhcp_option_update_95_option(strucWanConfig.mapt_support);
+
+            osp_ps_t *ps = NULL;
+            ps = osp_ps_open("MAPT_SUPPORT", OSP_PS_RDWR | OSP_PS_PRESERVE);
+            if (ps == NULL)
+            {
+                LOG(ERR, "Error saving new MAP-T support value");
+            }
+            if (!(osp_ps_set(ps, "MAPT_SUPPORT", maptm_get_supportValue(conf->value) ? "true" : "false",sizeof(maptm_get_supportValue(conf->value)))))
+            {
+                LOGE("Error saving new MAP-T support value");
+            }
+        }
+    }
+
+    if (mon->mon_type == OVSDB_UPDATE_DEL)
+    {
+        
+        LOGD("%s: node config entry deleted: module %s, key: %s, value: %s",
+                __func__, old_rec->module, old_rec->key, old_rec->value);
+
+        osp_ps_t *ps = NULL;
+        ps = osp_ps_open("MAPT_SUPPORT", OSP_PS_RDWR | OSP_PS_PRESERVE);
+        if (ps == NULL)
+        {
+            LOG(ERR, "Error saving new MAP-T support value");
+        }
+        if (!(osp_ps_set(ps, "MAPT_SUPPORT", "true", sizeof(maptm_get_supportValue(conf->value)))))
```

Comment:
`sizeof(maptm_get_supportValue(conf->value))` is `sizeof(bool)`, which is not what was intended.